### ISSUE TITLE
Clarification on target-domain exclusion during validation and checkpoint selection

### ISSUE THREAD

**Issue by Oracle1028** [AIRLABkhu/A2XP#5]
Hi, thank you for the previous clarification in Issue #2.

I understand that, when evaluating A2XP on the P domain, we should train expert prompts only for the A, C, and S domains, rather than training an expert prompt for P.

However, I am still confused about how the final target domain is excluded from validation and checkpoint selection in the released code.

According to the previous reply, when running `train_expert.py` for one specified domain, the other domains are used as validation data. For example, if P is the final target domain and we run:

```bash
python train_expert.py --dataset pacs --domain a
```

it seems that the A domain is used to train the A-domain expert prompt, while P, C, and S are used as validation data. Since the script saves the best expert checkpoint according to validation accuracy, this appears to use P-domain labels for model selection.

I could not find an argument in `train_expert.py` that specifies the final held-out target domain, so I am not sure how P is excluded from the validation set when evaluating on P.

I also noticed a related issue in `train.py`. The held-out target domain appears to be assigned to `valid_loader`, and the model checkpoint with the highest target-domain validation accuracy is saved as the best model.

I understand that the target domain may not be used for gradient updates. However, under a strict leave-one-domain-out domain generalization protocol, target-domain labels should also remain unavailable for validation, early stopping, hyperparameter selection, and checkpoint selection.

Could you please clarify the following?

1. When P is the final target domain, what are the exact commands or configurations used to train the A, C, and S expert prompts without using P-domain labels for validation or checkpoint selection?

2. For the reported results, was model selection performed using only source-domain validation splits?

3. Is there another script, commit, or configuration that implements this strict source-only model-selection protocol?

4. If the released code is intended only for a different evaluation setting, could you provide the recommended modifications for reproducing the strict domain generalization results?

I may be misunderstanding the intended usage of the repository, so I would greatly appreciate any clarification. Thank you for your time and for releasing the code.
